File: packages/js/src/tools/kyc/check-kyc-status.ts
import { z } from "zod";
import { NuggetsBaseTool } from "../base.js";
import type { KycResult } from "../../types.js";

const schema = z.object({
  sessionId: z.string().describe("The KYC session ID returned by initiate_kyc_verification"),
});

export class CheckKycStatus extends NuggetsBaseTool {
  name = "check_kyc_status";
  description = "Check the status of a KYC verification session. Returns status: \"pending\" (user has not yet completed), \"completed\" (verified), \"failed\" (verification failed), or \"expired\" (session timed out). If completed, includes the verified credentials.";
  schema = schema;

  async _call({ sessionId }: z.output<typeof schema>): Promise<string> {
    const result = await this.client.get<KycResult>(`/kyc/sessions/${encodeURIComponent(sessionId)}`);
    return JSON.stringify(result);
  }
}

File: packages/python/langchain_nuggets/tools/kyc/check_kyc_status.py
"""Check KYC status tool."""
from __future__ import annotations
from urllib.parse import quote

import json
from typing import Optional, Type

from langchain_core.callbacks import AsyncCallbackManagerForToolRun, CallbackManagerForToolRun
from pydantic import BaseModel, Field

from langchain_nuggets.tools.base import NuggetsBaseTool


class CheckKycStatusInput(BaseModel):
    sessionId: str = Field(description="The KYC session ID returned by initiate_kyc_verification")


class CheckKycStatus(NuggetsBaseTool):
    name: str = "check_kyc_status"
    description: str = 'Check the status of a KYC verification session. Returns status: "pending" (user has not yet completed), "completed" (verified), "failed" (verification failed), or "expired" (session timed out). If completed, includes the verified credentials.'
    args_schema: Type[BaseModel] = CheckKycStatusInput

    def _run(self, sessionId: str, run_manager: Optional[CallbackManagerForToolRun] = None) -> str:
        result = self.client.get(f"/kyc/sessions/{quote(sessionId, safe="")}")
        return json.dumps(result)

    async def _arun(self, sessionId: str, run_manager: Optional[AsyncCallbackManagerForToolRun] = None) -> str:
        result = await self.client.aget(f"/kyc/sessions/{quote(sessionId, safe="")}")
        return json.dumps(result)
